synapse/config/registration.py

```diff
@@ -127,14 +127,6 @@ def read_config(self, config, **kwargs):
             access_token_lifetime = self.parse_duration(access_token_lifetime)
         self.access_token_lifetime = access_token_lifetime
 
-        if session_lifetime is not None and access_token_lifetime is not None:
```

Comment:
Why were they incompatible before?

Reply:
Traditionally, `session_lifetime` controls the lifetime of 'regular' access tokens. As you can imagine, that's all you needed to do to get a session to end at a predetermined time.

When refresh tokens emerged, they work by creating new access tokens with (typically short) lifetimes. But you can use refresh tokens to get a new access token with a freshened lifetime. There was no mechanism in place to put a hard stop on a session's lifetime, since they could be refreshed perpetually.